### PR TITLE
[TACHYON-658] Allocator/evictor should extend BlockStoreEventListener…

### PR DESCRIPTION
**Pull request by miasantreble**
…Base

### CODE REVIEW

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 1]
```diff
@@ -95,18 +95,14 @@ public TieredBlockStore(TachyonConf tachyonConf) throws IOException {
     BlockMetadataManagerView initManagerView = new BlockMetadataManagerView(mMetaManager,
         Collections.<Integer>emptySet(), Collections.<Long>emptySet());
     mAllocator = AllocatorFactory.create(allocatorType, initManagerView);
-    if (mAllocator instanceof BlockStoreEventListener) {
-      registerBlockStoreEventListener((BlockStoreEventListener) mAllocator);
-    }
+    registerBlockStoreEventListener((BlockStoreEventListener) mAllocator);
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: Do you need to cast here?

File: servers/src/main/java/tachyon/worker/block/allocator/GreedyAllocator.java [hunk 2]
```diff
@@ -27,7 +28,7 @@
  * A greedy allocator that returns the first Storage dir fitting the size of block to allocate.
  * This class serves as an example how to implement an allocator.
  */
-public class GreedyAllocator implements Allocator {
+public class GreedyAllocator extends BlockStoreEventListenerBase implements Allocator {
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/allocator/GreedyAllocator.java]: Maybe it makes sense to make Evictor/Allocator extend the `BlockStoreEventListenerBase`? Its kind of weird to make all the evictors need to extend `BlockStoreEventListenerBase`.

Comment (miasantreble) [inline servers/src/main/java/tachyon/worker/block/allocator/GreedyAllocator.java]: @calvinjia you mean make Evictor/Allocator extend BlockStoreEventListener? I don't think they (interface) can extend BlockStoreEventListenerBase (class). In this case each implementation of Evitor/Allocator will have to implement all the methods specified by BlockStoreEventListener, right? Is that what we want? Shall we introduce an abstract class here?

Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/allocator/GreedyAllocator.java]: @miasantreble Originally I thought abstract class would be good, but now I think there's value in keeping it an interface. Do you have a strong opinion either way? If we go with interface, I think the ticket is no longer necessary.

Comment (miasantreble) [inline servers/src/main/java/tachyon/worker/block/allocator/GreedyAllocator.java]: @calvinjia +1 for keeping it an interface, I will close this PR and the JIRA, thanks!

Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/allocator/GreedyAllocator.java]: @miasantreble sounds good, thanks!

**Pull request closed without merging**